javaServices/coreJavaServices/src/main/java/joliex/db/DatabaseService.java

```diff
@@ -115,148 +128,102 @@ public void connect( Value request )
 					separator = ";";
 					databaseName = "databaseName=" + databaseName;
-				} else if( "as400".equals( driver ) ) {
-					Class.forName( "com.ibm.as400.access.AS400JDBCDriver" );
-				} else if( "derby_embedded".equals( driver ) ) {
-					Class.forName( "org.apache.derby.jdbc.EmbeddedDriver" );
+					break;
+				case "as400":
+					driverClass = "com.ibm.as400.access.AS400JDBCDriver";
+					break;
+				case "derby_embedded":
+					driverClass = "org.apache.derby.jdbc.EmbeddedDriver";
 					isEmbedded = true;
 					driver = "derby";
-				} else if( "hsqldb_hsql".equals( driver )
-					|| "hsqldb_hsqls".equals( driver )
-					|| "hsqldb_http".equals( driver )
-					|| "hsqldb_https".equals( driver ) ) {
-					Class.forName( "org.hsqldb.jdbcDriver" );
-				} else if( "hsqldb_embedded".equals( driver ) ) {
-					Class.forName( "org.hsqldb.jdbcDriver" );
+					break;
+				case "hsqldb_embedded":
 					isEmbedded = true;
 					driver = "hsqldb";
-				} else if( "db2".equals( driver ) ) {
-					Class.forName( "com.ibm.db2.jcc.DB2Driver" );
-				} else {
+				case "hsqldb_hsql":
+				case "hsqldb_hsqls":
+				case "hsqldb_http":
+				case "hsqldb_https":
+					driverClass = "org.hsqldb.jdbc.JDBCDriver";
+					break;
+				case "db2":
+					driverClass = "com.ibm.db2.jcc.DB2Driver";
+					break;
+				default:
 					throw new FaultException( "InvalidDriver", "Unknown type of driver: " + driver );
 				}
-			} else {
-				Class.forName( driverClass );
 			}
+			Class.forName( driverClass );
 
 			if( isEmbedded ) {
 				connectionString = "jdbc:" + driver + ":" + databaseName;
 				if( !attributes.isEmpty() ) {
 					connectionString += ";" + attributes;
 				}
-				if( "hsqldb".equals( driver ) ) {
-					connection = DriverManager.getConnection(
-						connectionString,
-						username,
-						password );
-				} else {
-					connection = DriverManager.getConnection( connectionString );
+				connectionPool = _createDataSource();
+				if( !"hsqldb".equals( driver ) ) { // driver == sqlite || driver == derby_embedded
+					connectionPool.setUsername( null );
+					connectionPool.setPassword( null );
 				}
 			} else {
 				if( driver.startsWith( "hsqldb" ) ) {
 					connectionString = "jdbc:" + driver + ":" + driver.substring( driver.indexOf( '_' ) + 1 ) + "//"
 						+ host + (port.isEmpty() ? "" : ":" + port) + separator + databaseName;
 				} else {
-					connectionString =
-						"jdbc:" + driver + "://" + host + (port.isEmpty() ? "" : ":" + port) + separator + databaseName;
+					connectionString = "jdbc:" + driver + "://" + host + (port.isEmpty() ? "" : ":" + port) + separator
+						+ databaseName;
 				}
 				if( encoding.isPresent() ) {
 					connectionString += "?characterEncoding=" + encoding.get();
 				}
-				connection = DriverManager.getConnection(
-					connectionString,
-					username,
-					password );
+				connectionPool = _createDataSource();
 			}
 
-			if( connection == null ) {
```

Comment:
Using the connection pool, this check is unnecessary. The connection pool will fail to create and throw a SQLTransientConnectionException if it cannot validate any connections.